### PR TITLE
System information v2

### LINKED ISSUE #14: Return global information about the system
Add the functionality of returning information about the system:
- Uptime
- Number of users (not needed for now since we have not added users to the application)
- Number of URL shortened
- Number of clicks (URL redirects)
- Memory being used by the system
- Available memory of the system

### PR DESCRIPTION
**Pull request by luisgg98**
I've added actuator to the project, it provides the information we committed on the project contract.
I followed these tutorials:
https://www.baeldung.com/spring-boot-actuators
https://www.baeldung.com/spring-boot-actuators#boot-2x-actuator

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -3,4 +3,32 @@ spring.datasource.platform=hsqldb
...
+management.endpoints.web.exposure.include=metrics,database
+
+
+
+
+
+
+#endpoints.health.sensitive=true
+#management.security.enabled=false
+# There is no security
```
Comment (AlbertoCalvoRubio) [inline src/main/resources/application.properties]: Is this necessary?

File: src/main/java/urlshortener/service/ShortURLService.java [hunk 2]
```diff
@@ -38,4 +38,8 @@ public ShortURL save(String url, String sponsor, String ip) {
             .build();
     return shortURLRepository.save(su);
   }
+
+  public Long getTotalURL() {
+    return shortURLRepository.count();
+  } // NOSONAR
```
Comment (AlbertoCalvoRubio) [inline src/main/java/urlshortener/service/ShortURLService.java]: Why is it necessary here? What do you want to surpress?

**Review by AlbertoCalvoRubio — CHANGES_REQUESTED** [review]
Check the comments and refactor that lines if it is necessary

File: src/main/java/urlshortener/service/ShortURLService.java (hunk 2 above)
Comment (luisgg98) [reply src/main/java/urlshortener/service/ShortURLService.java]: image
Sorry I thought the whole function was covered by the comment

File: src/main/resources/application.properties (hunk 1 above)
Comment (luisgg98) [reply src/main/resources/application.properties]: No, I'll erase it

File: src/main/java/urlshortener/domain/UrlStatus.java [hunk 3]
```diff
@@ -4,7 +4,8 @@
   OK("OK"),
   UNSAFE("Unsafe url"),
   INVALID("Invalid url"),
-  UNREACHABLE("Unreachable url");
+  UNREACHABLE("Unreachable url"),
+  IDNOTFOUND("ID of shortened URL not found");
```
Comment (AlbertoCalvoRubio) [inline src/main/java/urlshortener/domain/UrlStatus.java]: This isn't a URL Status. It makes no sense to put this here.

File: src/main/java/urlshortener/service/ClickService.java [hunk 4]
```diff
@@ -25,4 +24,8 @@ public void saveClick(String hash, String ip) {
             ? "[" + hash + "] saved with id [" + cl.getId() + "]"
             : "[" + hash + "] was not saved");
   }
+
+  public Long getTotalClick() { // NOSONAR
+    return clickRepository.count(); // NOSONAR
```
Comment (AlbertoCalvoRubio) [inline src/main/java/urlshortener/service/ClickService.java]: Why //NOSONAR?

File: src/test/java/urlshortener/web/UrlShortenerTests.java [hunk 5]
```diff
@@ -132,6 +133,7 @@ public void thatShortenerFailsIfTheURLisNotSafe() throws Exception {
         .andExpect(status().isBadRequest());
   }
 
+  //ok
```
Comment (AlbertoCalvoRubio) [inline src/test/java/urlshortener/web/UrlShortenerTests.java]: ?

**Review by AlbertoCalvoRubio — CHANGES_REQUESTED** [review]
Read comments below and answer the questions

File: src/main/java/urlshortener/service/ClickService.java (hunk 4 above)
Comment (luisgg98) [reply src/main/java/urlshortener/service/ClickService.java]: image
Same mistake again sorry

File: src/test/java/urlshortener/web/UrlShortenerTests.java (hunk 5 above)
Comment (luisgg98) [reply src/test/java/urlshortener/web/UrlShortenerTests.java]: I was checking one by one if the tests were working I didn't realize I left that comment

File: src/main/java/urlshortener/domain/UrlStatus.java (hunk 3 above)
Comment (luisgg98) [reply src/main/java/urlshortener/domain/UrlStatus.java]: So I'll create a local variable instead

**Review by AlbertoCalvoRubio — APPROVED** [review]

**Pull request merged**